src/components/buttons/button.tsx

```diff
@@ -1,29 +1,45 @@
 import React from 'react';
 import styled from 'styled-components';
-import { TextStyles } from '../typography/typography-v2';
+import { Icon, IconSize } from '../icon/icon';
+import { resetButton } from '../utils';
+import { lighten } from 'polished';
 
 const StyledButton = styled.button`
-  height: 40px;
+  ${resetButton};
+  color: #ffffff;
+  font-size: 16px;
+  font-weight: bold;
+  line-height: 40px;
+  border-radius: 30px;
+  vertical-align: middle;
+  cursor: pointer;
 
   background: linear-gradient(275.41deg, #543fd7 0%, #2756fd 100%);
-  border-radius: 30px;
-  border: none;
 
   &:hover {
     background: #3e61ee;
   }
+
+  &:active {
+    background: ${lighten(0.1, '#3e61ee')};
+  }
```

Comment:
I'm sorry, this might be beyond this PR of introducing the icon. I started adapting the button to make it work with the icon component and I got carried away a little bit by adding some active state and the cursor. Those are minimal changes and I didn't want to revert that useful stuff here.